CMakeLists.txt

```diff
@@ -118,16 +123,57 @@ add_definitions(-DBOOST_NO_CXX11_SCOPED_ENUMS)
 add_definitions(-DBOOST_ALLOW_DEPRECATED_HEADERS)
 add_definitions(-DBOOST_BIND_GLOBAL_PLACEHOLDERS)
 
+# resolve vendored libraries
+set(date_include_dir ${VALHALLA_SOURCE_DIR}/third_party/date/include)
```

Comment:
this is the main changes, defaulting to vendored paths but trying to change them to system paths if opted in with `PREFER_SYSTEM_DEPS`